insteon_mqtt/device/KeypadLinc.py

```diff
@@ -289,115 +261,113 @@ def on(self, group=0, level=None, mode=on_off.Mode.NORMAL, reason="",
                     # level, go to full brightness.
                     level = 0xff
 
-        assert 1 <= group <= 9
-        assert 0 <= level <= 0xff
-        assert isinstance(mode, on_off.Mode)
+        mode, transition = self.mode_transition_supported(mode, transition)
+
+        cmd1 = on_off.Mode.encode(True, mode)
+        cmd2 = on_off.Mode.encode_cmd2(True, mode, level, transition)
+        return (cmd1, cmd2)
+
+    #-----------------------------------------------------------------------
+    def cmd_off_values(self, mode, transition, group):
+        """Calculate Cmd Values for Off
+
+        Args:
+          mode (on_off.Mode): The type of command to send (normal, fast, etc).
+          transition (int): Ramp rate for the transition in seconds.
+        Returns
+          cmd1, cmd2 (int): Value of cmds for this device.
+        """
+        # Ignore RAMP mode / transition if command not supported
+        mode, transition = self.mode_transition_supported(mode, transition)
+        cmd1 = on_off.Mode.encode(False, mode)
+        cmd2 = on_off.Mode.encode_cmd2(True, mode, 0, transition)
+        return (cmd1, cmd2)
+
+    #-----------------------------------------------------------------------
+    def on(self, group=0x01, level=None, mode=on_off.Mode.NORMAL, reason="",
```

Comment:
It looks like you've changed the default group value from 0 to 1 for the on, off, and set functions.  This appears as if it might break deployments with detached loads, because _input_set_level() (in insteon_mqtt/mqtt/KeypadLinc.py) calls self.device.set() without specifying a group number.  So with the old code, it would be calling set with a default value of 0 (becoming group=9, with detached loads), and with the new code it would get a default of 1 (which explicitly indicates the load-less button 1, when the load is detached).

... actually the default value of set() looks like it was changed as part of #325, FWIW.  Either way, it could probably be solved by explicitly specifying group=0 when calling self.device.set() in insteon_mqtt/mqtt/KeypadLinc.py :: _input_set_level().

Reply:
Fixed

Reply:
I will write a unit test to figure this out.

Reply:
I change KPL back to default to 0x00 which returns it back to the prior usage and added a unit test for level with a detached load.

Personally, using group=0 to mean either group 1 or 9 depending on the device setup seems a bit icky to me.  If anyone ever cleans up the KPL device that might be worth looking into.